lib/src/dom/streams.dart

```diff
@@ -198,22 +635,102 @@ extension type UnderlyingSink._(JSObject _) implements JSObject {
   external set type(JSAny? value);
   external JSAny? get type;
 }
+
+/// The **`WritableStreamDefaultWriter`** interface of the
+/// [Streams API](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API)
+/// is the object returned by [WritableStream.getWriter] and once created locks
+/// the writer to the `WritableStream` ensuring that no other streams can write
+/// to the underlying sink.
 extension type WritableStreamDefaultWriter._(JSObject _) implements JSObject {
   external factory WritableStreamDefaultWriter(WritableStream stream);
 
+  /// The **`abort()`** method of the
+  /// [WritableStreamDefaultWriter] interface aborts the stream, signaling that
+  /// the producer can no longer successfully write to the stream and it is to
+  /// be immediately
+  /// moved to an error state, with any queued writes discarded.
+  ///
+  /// If the writer is active, the `abort()` method behaves the same as that for
+  /// the associated stream ([WritableStream.abort]). If not, it returns a
+  /// rejected promise.
   external JSPromise<JSAny?> abort([JSAny? reason]);
+
+  /// The **`close()`** method of the
+  /// [WritableStreamDefaultWriter] interface closes the associated writable
+  /// stream.
+  ///
+  /// The underlying sink will finish processing any previously-written chunks,
+  /// before
```

Comment:
It feels like there's a bug in the wrapping algorithm here.

Reply:
I looked into this; it looks like there's a newline after the `before` in the source material.

We're currently respecting newlines in the source (instead of completely re-flowing everything). From memory, I think this made parsing some of the markdown constructs simpler (code blocks? blockquotes?).

I think there's an argument for re-flowing from scratch, as that would make the generated dartdocs look better. I'd like to hold off on any work in that direction for future PRs however. We'd want to move from a quick-and-dirty parser to something that understood markdown. And, we could even choose to push any impl. further into tooling - being able to format dartdoc comments to 80-cols would be a very nice feature in general.

cc @srawlins who has thought a bit about the work to format a chunk of markdown.

Reply:
> I think there's an argument for re-flowing from scratch, as that would make the generated dartdocs look better.

The generated dartdocs should not be affected, if I'm looking at this right. The consecutive lines, "The underlying ...", "before", "invoking the ..." are all in one Markdown paragraph. The newlines are considered soft.

> being able to format dartdoc comments to 80-cols would be a very nice feature in general.

Not sure if there are command line tools for this, but in vim you can `:gq` 😁  I think there is a VS Code hotkey / plugin as well. Obviously doing it from `dart format` or a `dart format` wrapper would be really boss.

Reply:
Got it - adding a TODO and then coming back to it is fine. It's possible we might me able to detect where code blocks begin and end to avoid removing newlines from those.

Reply:
@srawlins Oh btw, I discovered https://marketplace.visualstudio.com/items?itemName=stkb.rewrap thanks to your comment and you've saved me hours of manual rewrapping pain that I've done in the past. 🙏